teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -90,23 +102,28 @@ export class MapsService {
   async addNodesFromClient(
     mapId: string,
     clientNodes: IMmpClientNode[]
-  ): Promise<MmpNode[]> {
+  ): Promise<MmpNode[] | undefined> {
     const mmpNodes = clientNodes.map(x => mapClientNodeToMmpNode(x, mapId))
     return await this.addNodes(mapId, mmpNodes)
   }
 
   async addNodes(
     mapId: string,
     nodes: Partial<MmpNode>[]
-  ): Promise<MmpNode[]> {
-    if (!mapId || nodes.length === 0) return Promise.reject()
+  ): Promise<MmpNode[] | undefined> {
+    if (!mapId || nodes.length === 0) {
+      this.logger.warn(`Required arguments mapId or nodes not supplied to addNodes()`);
+      return;
+    }
 
       const reducer = async (previousPromise: Promise<MmpNode[]>, node: MmpNode): Promise<MmpNode[]> => {
         const accCreatedNodes = await previousPromise;
         if (await this.validatesNodeParentForNode(mapId, node)) {
           try {
             const newNode = await this.addNode(mapId, node);
-            return accCreatedNodes.concat([newNode]);
+            if (newNode) {
+              return accCreatedNodes.concat([newNode]);
```

Comment:
be careful to not break the reduce function here - I think a return is missing in the case `newNode` is missing. Maybe just move the return in the catch part one level out of the current scope?

Reply:
I'll return an empty array just in case.

Reply:
why an empty array? this breaks the whole reduce function due to one potentially broken node. You need to return `accCreatedNodes` here like it is already done in the catch block. You can already reuse the return there if you move it one level down out of the catch block to make it the last line in the function call.

Reply:
Yeah good point :D It should now return accCreatedNodes properly.